### PR TITLE
[FEATURE] Add lowercaseKeys

### PR DESCRIPTION
**Pull request by Chalarangelo**
## Description

Creates a new object from the specified object, where all the keys are in lowercase.

## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: snippets/lowercaseKeys.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+```js
+const lowercaseKeys = obj =>
+  Object.keys(obj).reduce((acc,key) => {acc[key.toLowerCase()] = obj[key]; return acc;},{});
+```
+
+```js
+let myObj = {Name: 'Adam', sUrnAME: 'Smith'};
+let myObjLower = lowercaseKeys(myObj);
+console.log(myObj); // {Name: 'Adam', sUrnAME: 'Smith'};
+console.log(myObjLower); // {name: 'Adam', surname: 'Smith'};
```
Comment (Pl4gue) [inline snippets/lowercaseKeys.md]: I would shorten the example to 
```js
 let myObj = {Name: 'Adam', sUrnAME: 'Smith'}; 
 let myObjLower = lowercaseKeys(myObj); // {name: 'Adam', surname: 'Smith'};
```

Comment (Chalarangelo) [reply snippets/lowercaseKeys.md]: I only have the logging to showcase that these are different objects. Do you think it's unnecessary?

Comment (atomiks) [reply snippets/lowercaseKeys.md]: Always prefer `const` over `let` unless you're reassigning it btw.

Comment (Chalarangelo) [reply snippets/lowercaseKeys.md]: Actually using `const` will be enough and I can remove the logging. Makes sense!

**Pull request merged**